src/View/ThemeResourceLoader.php

```diff
@@ -3,20 +3,30 @@
 namespace SilverStripe\View;
 
 use InvalidArgumentException;
+use Psr\SimpleCache\CacheInterface;
+use SilverStripe\Core\Flushable;
+use SilverStripe\Core\Injector\Injector;
 use SilverStripe\Core\Manifest\ModuleLoader;
 use SilverStripe\Core\Path;
 
 /**
  * Handles finding templates from a stack of template manifest objects.
  */
-class ThemeResourceLoader
+class ThemeResourceLoader implements Flushable
 {
 
     /**
      * @var ThemeResourceLoader
      */
     private static $instance;
 
+    /**
+     * Internal memory cache for large sets of repeated calls
+     *
+     * @var array
+     */
+    protected static $cacheData = [];
```

Comment:
@robbieaverill should this have been removed?

Reply:
Yep, soz. PR incoming

Reply:
#8463